react/features/mobile/call-integration/middleware.js

```diff
@@ -248,8 +238,39 @@ function _conferenceWillJoin({ getState }, next, action) {
                     MEDIA_TYPE.AUDIO);
 
             // eslint-disable-next-line object-property-newline
-            CallKit.updateCall(conference.callUUID, { displayName, hasVideo });
-            CallKit.setMuted(conference.callUUID, muted);
+            CallIntegration.updateCall(
+                conference.callUUID,
+                {
+                    displayName,
+                    hasVideo
+                });
+            CallIntegration.setMuted(conference.callUUID, muted);
+        })
+        .catch(error => {
+            // Currently this error code is emitted only by Android.
+            if (error.code === 'CREATE_OUTGOING_CALL_FAILED') {
+                // We're not tracking the call anymore - it doesn't exist on
+                // the native side.
+                delete conference.callUUID;
+                dispatch({
+                    type: CONFERENCE_FAILED,
+                    conference,
+                    error: {
+                        name: 'REQUEST_DENIED_BY_THE_SYSTEM',
+                        recoverable: false
+                    }
+                });
+                dispatch(appNavigate(undefined));
```

Comment:
won't this trigger 2 events? First this CONFERENCE_FAILED, then CONFERENCE_ENDED, when appNavigate does its thing.

I'd remove the CONFERENCE FAILED, since the alert will already tell you what this is about.

WDYT?

Reply:
yes